### DISCUSSION TITLE
Query about Mel spectogram normalisation

### DISCUSSION THREAD

**Discussion by Neuhaus24** [openai/whisper#902] [Q&A] [answered]
In lines 121-124 of audio.py the mel_spectogram has a log taken and then sort of normalised but not around zero.  I'm familar with the practice of normalising data around zero in neural nets to maximise gradient but am unsure as to why here it is being normalised around a quarter of the maximum value.  Just wondering if there is a specific reason or just that experimentally that's what worked

**Comment by jongwook** [comment, marked as answer]
The main reason for L122 was that during training we used [`librosa.power_to_db` which does this with the default argument `top_db=80.0`](https://librosa.org/doc/latest/generated/librosa.power_to_db.html), and we wanted to replicate this without having to depend on librosa.

L123 was to put the numbers roughly roughly into [-1, 1], but as you noticed it doesn't strictly put those within [-1, 1]. In practice, this shouldn't matter much as long as we do the same between training and inference.

https://github.com/openai/whisper/blob/4e635c66441dfae3c6c72365f802293afffca3ef/whisper/audio.py#L122-L123

As for why that became the default argument for `librosa.power_to_db`, maybe @bmcfee (hi!!) can answer?

**Reply by bmcfee** [reply]
> As for why that became the default argument for `librosa.power_to_db`, maybe @bmcfee (hi!!) can answer?

:wave:

This was carried over for compatibility with a reference implementation in matlab.  Details are a bit hazy 10 years later, but IIRC it was the mel spectrogram implementation extracted from Dan Ellis's [beat tracker code](https://labrosa.ee.columbia.edu/projects/coversongs/tempo.m.html).  There were several slightly different implementations kicking around, even within our reference code base, and we settled on that one I suppose because it's both numerically well-behaved (bounded) and it makes some intuitive sense to limit attention to something near human perceptual limits.
